src/common/services/session/session.service.spec.js

```diff
@@ -357,40 +316,108 @@ describe('session service', function () {
       })
     })
 
-    it('should redirect to Okta from the login screen if the login has not yet happened', done => {
-      jest.spyOn($location, 'path').mockReturnValue('/sign-in.html')
-      sessionService.authClient.setLoginRedirect(true)
-      sessionService.authClient.setAuthenticated(false)
-      sessionService.authClient.shouldSucceed()
-      sessionService.handleOktaRedirect().subscribe(() => {
-        expect(sessionService.authClient.token.getWithRedirect).toHaveBeenCalled()
-        expect($window.sessionStorage.getItem(locationSearchOnLogin)).toEqual('?ga=111111&query=test&anotherQuery=00000')
-        expect($window.sessionStorage.getItem(locationOnLogin)).toEqual(null)
-        done()
-      })
-    })
-
-    it('should redirect to Okta from page other than login if the login has not yet happened', done => {
-      jest.spyOn($location, 'path').mockReturnValue('/search-results.html')
-      sessionService.authClient.setLoginRedirect(true)
-      sessionService.authClient.setAuthenticated(false)
-      sessionService.authClient.shouldSucceed()
-      sessionService.handleOktaRedirect().subscribe(() => {
-        expect(sessionService.authClient.token.getWithRedirect).toHaveBeenCalled()
-        expect($window.sessionStorage.getItem(locationSearchOnLogin)).toEqual(null)
-        expect($window.sessionStorage.getItem(locationOnLogin)).toEqual('https://URL.org?utm_source=text')
-        done()
-      })
-    })
 
     it('should ignore a non-login attempt', done => {
-      sessionService.authClient.setLoginRedirect(false)
+      jest.spyOn(sessionService.authClient, 'isAuthenticated').mockResolvedValueOnce(false)
+      jest.spyOn(sessionService.authClient.token, 'parseFromUrl').mockResolvedValue({})
       sessionService.handleOktaRedirect().subscribe((data) => {
         expect(sessionService.authClient.token.parseFromUrl).not.toHaveBeenCalled()
         expect(data).toEqual(false)
         done()
       })
     })
+
+    describe('successful handleOktaRedirect()', () => {
+      const accessToken = 'accessToken'
+      let isAuthenticatedSpy
+      beforeEach(() => {
```

Comment:
You might want to add a spy for the `signIn` function since that is tested elsewhere.